### ISSUE TITLE
GPU Training Cancelled

### ISSUE THREAD

**Issue by wes-baldwin** [dotnet/machinelearning-modelbuilder#1184]
**System Information (please complete the following information):**
 - Model Builder Version: Latest version ML.NET Model Builder GPU Support (Preview)
 - Visual Studio Version: 2019

**Describe the bug**
- On which step of the process did you run into an issue: Training
- Clear description of the problem: After about 30 minutes the training failed with error Training Cancelled. The stack trace from the console is below...
- [Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Bottleneck Computation, Dataset used:      Train, Image Index: 10439
[Source=AutoML, Kind=Error] Pipeline crashed: xf=ValueToKeyMapping{ col=Label:Label} xf=RawByteImageLoading{ col=ImageSource_featurized:ImageSource imageFolder=} xf=ColumnCopying{ col=Features:ImageSource_featurized} tr=ImageClassification{} xf=KeyToValueMapping{ col=PredictedLabel:PredictedLabel} cache=- . Exception: Tensorflow.TensorflowException: Blas GEMM launch failed : a.shape=(10, 1001), b.shape=(10, 5), m=1001, n=5, k=10
	 [[{{node train/gradients/final_retrain_ops/Wx_plus_b/MatMul_grad/MatMul_1}}]]
   at Tensorflow.Status.Check(Boolean throwException)
   at Microsoft.ML.TensorFlow.TensorFlowUtils.Runner.Run()
   at Microsoft.ML.Vision.ImageClassificationTrainer.TrainAndEvaluateClassificationLayerCore(Int32 epoch, Single learningRate, Int32 featureFileStartOffset, ImageClassificationMetrics metrics, Int64[] labelTensorShape, Int64[] featureTensorShape, Int32 batchSize, Stream trainSetLabelReader, Stream trainSetFeatureReader, Byte[] labelBufferBytes, Byte[] featuresBufferBytes, Int32 labelBufferSizeInBytes, Int32 featureBufferSizeInBytes, Int32 featureFileRecordSize, LearningRateScheduler learningRateScheduler, DnnTrainState trainState, Runner runner, IntPtr featureBufferPtr, IntPtr labelBufferPtr, Action`2 metricsAggregator)
   at Microsoft.ML.Vision.ImageClassificationTrainer.TrainAndEvaluateClassificationLayer(String trainBottleneckFilePath, String validationSetBottleneckFilePath)
   at Microsoft.ML.Vision.ImageClassificationTrainer.TrainModelCore(TrainContext trainContext)
   at Microsoft.ML.Trainers.TrainerEstimatorBase`2.TrainTransformer(IDataView trainSet, IDataView validationSet, IPredictor initPredictor)
   at Microsoft.ML.Data.EstimatorChain`1.Fit(IDataView input)
   at Microsoft.ML.Data.EstimatorChain`1.Fit(IDataView input)
   at Microsoft.ML.AutoML.RunnerUtil.TrainAndScorePipeline[TMetrics](MLContext context, SuggestedPipeline pipeline, IDataView trainData, IDataView validData, String groupId, String labelColumn, IMetricsAgent`1 metricsAgent, ITransformer preprocessorTransform, FileInfo modelFileInfo, DataViewSchema modelInputSchema, IChannel logger)
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Channel started
[Source=ImageClassificationTrainer; Ensuring meta files are present., Kind=Trace] Channel started
[Source=ImageClassificationTrainer; Ensuring meta files are present., Kind=Trace] Channel disposed
[Source=ImageClassificationTrainer; Ensuring meta files are present., Kind=Trace] Channel finished. Elapsed 00:00:00.0001597.
Training cancelled

**To Reproduce**
1. Download the sample dataset from kaggle @ https://www.kaggle.com/puneet6060/intel-image-classification
2. Step thru the model builder UI (selecting the GPU environment) and select the train folder from the extracted archive.zip from kaggle
3. Click Train
4. After about 20 minutes (for my machine) it will begin displaying image indexes in the console.
5. 5-10 minutes later the stated error occurs.

**Expected behavior**
I expected the training to complete

**Screenshots**
![Capture1](https://user-images.githubusercontent.com/5668004/105260406-63389d80-5b53-11eb-8657-4a626b3355fa.PNG)
![Capture2](https://user-images.githubusercontent.com/5668004/105260416-66338e00-5b53-11eb-91aa-b725ce9655d4.PNG)

**Additional context**
I installed all software as stated on https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/install-gpu-model-builder.
My machine info:
![Capture3](https://user-images.githubusercontent.com/5668004/105260814-143f3800-5b54-11eb-9c98-71fd06b4692b.PNG)
![Capture4](https://user-images.githubusercontent.com/5668004/105260826-17d2bf00-5b54-11eb-8888-d94cb8d59918.PNG)

**Comment by JakeRadMSFT** [comment]
@wes-baldwin I just finished running this and I don't think the above will resolve the issue for this dataset.

We're looking into why it's canceling early

**Comment by wes-baldwin** [comment]
Thanks for the response. I’ve since moved on to other projects. I’m don’t require a resolution. I really posted this more for the benefit of others.
